.github/workflows/cd.yaml

```diff
@@ -0,0 +1,28 @@
+name: CD
+on:
+  push:     
+    branches:
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v14.1
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+      - uses: cachix/cachix-action@v10
+        with:
+          name: kaiko-ai
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+      - name: Import GPG Key
+        uses: crazy-max/ghaction-import-gpg@v1
+        env:
+          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+          PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+      - name: Release Snapshot
+        run: nix-shell --run "sbt 'publishSigned'"
```

Comment:
does this update version.sbt ? or push commits?

Reply:
If the version.sbt has `SNAPSHOT` suffix, this will build the .jar, sign it with the GPG key and push it to the maven snapshot repository. If there is no `SNAPSHOT` suffix, it will build the jar to a local folder and sign it with the GPG key. But as long as we keep the release a manual step, the `SNAPSHOT` suffix will always be there so it will push to the snapshot repository.

Version is not updated, and no commits are created/pushed

Reply:
Some more info about the plugin: https://github.com/xerial/sbt-sonatype#sbt-sonatype-plugin